build-system/common/ci.js

```diff
@@ -87,49 +48,36 @@ function isCiBuild() {
 }
 
 /**
- * Returns true if this is a PR build.
- * @return {boolean}
- */
-function isPullRequestBuild() {
-  return serviceFunctionMap['isPullRequestBuild']();
-}
-
-/**
- * Returns true if this is a push build.
- * @return {boolean}
+ * Returns a functions that tests if a build type is active.
+ * @param {string} desiredBuildType
+ * @return {Function}
  */
-function isPushBuild() {
-  return serviceFunctionMap['isPushBuild']();
+function testBuildType(desiredBuildType) {
+  return () => buildType === desiredBuildType;
 }
 
 /**
- * Returns the name of the PR branch.
+ * Returns a CI build variable, reporting an error if it's not a PR build.
+ * @param {Function} getter
  * @return {string}
  */
-function ciPullRequestBranch() {
-  if (!isPullRequestBuild()) {
-    log(red('ERROR:'), 'Not a PR build, PR branch is undefined.');
+function getPrBuildVar(getter) {
+  if (isPullRequestBuild()) {
+    log(red('ERROR:'), 'Not a PR build, value is undefined.');
     return '';
   }
-  return serviceFunctionMap['ciPullRequestBranch']();
-}
-
-/**
- * Returns the commit SHA being tested by the PR build.
- * @return {string}
- */
-function ciPullRequestSha() {
-  if (!isPullRequestBuild()) {
-    log(red('ERROR:'), 'Not a PR build, PR SHA is undefined.');
-    return '';
-  }
-  return serviceFunctionMap['ciPullRequestSha']();
+  return getter();
 }
 
 module.exports = {
-  ciPullRequestBranch,
-  ciPullRequestSha,
   isCiBuild,
   isPullRequestBuild,
   isPushBuild,
+
+  isTravisBuild: testBuildType('travis'),
+  isGithubActionsBuild: testBuildType('github_actions'),
+  isCircleciBuild: testBuildType('circleci'),
+
+  ciPullRequestBranch: getPrBuildVar(pullRequestBranch),
```

Comment:
These could be expanded pretty easily to support fields like build/job URL and commit sha across the different build types